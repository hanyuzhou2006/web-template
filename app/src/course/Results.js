import React, { Component } from 'react';
import Item from './Item';
export default class Results extends Component {

    render() {
        const { results = [], active, onActive } = this.props;
        return (
            <div className="entities">
                {results.map((v, i) => {
                    return <Item avatar={v.avatar} name={v.name} key={i} actived={active === i ? true : false}
                        onClick={() => { onActive(i) }}
                    />
                })}
            </div>
        )
    }
}