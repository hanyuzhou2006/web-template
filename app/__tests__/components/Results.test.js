import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Results from '../../src/course/Results';

const results = [
    { avatar: "/images/1.jpg", name: "Liuxia@sansi.com" },
    { avatar: "", name: "Hello World" },
]
const active = 0;

test('render', () => {
    const component = renderer.create(
        <Results results={results} active={active} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

 

test('vdom render', () => {
    const doms = mount(
         <Results results={results} active={active} />
    );
    const entities = doms.find('.entity');
    expect(entities.length).toBe(2);
    const e_active = entities.at(0);
    expect(e_active.find('.name').text()).toBe('Liuxia@sansi.com');
    
})




