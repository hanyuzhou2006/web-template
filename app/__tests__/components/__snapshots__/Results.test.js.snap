exports[`test render 1`] = `
<div
  className="entities">
  <div
    className="entity active"
    onClick={[Function]}>
    <span
      className="p1">
      <div
        className="person-icon"
        style={
          Object {
            "backgroundImage": "url(/images/1.jpg)",
          }
        } />
    </span>
    <span
      className="p2">
      <span
        className="name">
        Liuxia@sansi.com
      </span>
    </span>
  </div>
  <div
    className="entity "
    onClick={[Function]}>
    <span
      className="p1">
      <div
        className="person-icon"
        style={Object {}} />
    </span>
    <span
      className="p2">
      <span
        className="name">
        Hello World
      </span>
    </span>
  </div>
</div>
`;
